tests/console/docker_image.pm

```diff
@@ -7,69 +7,70 @@
 # notice and this notice are preserved.  This file is offered as-is,
 # without any warranty.
 
-# Summary: Test installation and running of the docker image for this snapshot
-# Maintainer: Fabian Vogt <fvogt@suse.com>
+# Summary: Test installation and running of the docker image from the registry for this snapshot
+# Maintainer: Pavel Dostál <pdostal@suse.cz>
 
 use base 'consoletest';
 use testapi;
 use utils;
 use strict;
-use version_utils qw(is_leap is_tumbleweed);
+use registration; qw(add_suseconnect_product);
+use version_utils qw(is_sle);
 
 sub run {
-    select_console 'root-console';
+    select_console "root-console";
 
-    my $version = get_required_var('VERSION');
+    my $version     = get_required_var("VERSION");
+    my $SCC_REGCODE = get_required_var("SCC_REGCODE");
 
-    my $image_name;
-    my $image_path;
-
-    if (is_tumbleweed) {
-        $image_name = 'opensuse/tumbleweed:current';
-        $image_path = '/usr/share/suse-docker-images/native/*-image*.tar.xz';
-
-        # For Tumbleweed, the image is wrapped inside an RPM
-        zypper_call "in docker opensuse-tumbleweed-image";
+    if (script_run("SUSEConnect --status-text") != 0) {
+        assert_script_run("SUSEConnect --cleanup");
+        assert_script_run("SUSEConnect -r $SCC_REGCODE");
+        add_suseconnect_product("sle-module-containers", substr($version, 0, 2));
     }
-    elsif (is_leap('15.0+')) {
-        $image_name = "opensuse-leap-$version:current";
-        $image_path = "/usr/share/suse-docker-images/native/opensuse-leap${version}-image.tar.xz";
 
-        # For Leap, the docker image is the ISO asset
-        my $image_filename = get_required_var('ISO');
-        $image_filename =~ s/^.*\///;
-        my $image_url = autoinst_url("/assets/other/$image_filename");
-        assert_script_run "curl $image_url --create-dirs -o $image_path";
+    my $image_name;
+    if (is_sle("=12-SP3")) {
+        $image_name = "registry.suse.de/suse/sle-12-sp3/update/products/casp30/container/sles12:sp3";
+    }
+    elsif (is_sle("=15")) {
+        $image_name = "registry.suse.de/suse/sle-15/update/cr/images/suse/sle15:current";
     }
     else {
-        die 'Only know about Tumbleweed and Leap 15.0+ docker images';
+        die("This test only works at SLE12SP3 and SLE15.");
     }
 
-    # Start the docker daemon, normally done by previous test modules already
-    systemctl 'start docker';
-    systemctl 'status docker';
-    assert_script_run 'docker info';
+    # Allow our internal 'insecure' registry
+    zypper_call("in docker");
+    assert_script_run("mkdir -p /etc/docker");
+    assert_script_run('cat /etc/docker/daemon.json; true');
```

Comment:
why ;true ? can this fail ?

Reply:
Yes, this file is not present after installation. But since there're more Docker tests I wanted to at least `cat` it before rewriting.